src/pages/Kavita/main.ts

```diff
@@ -0,0 +1,212 @@
+import { pageInterface } from '../pageInterface';
+import { pathToUrl, urlToSlug } from '../../utils/slugs';
+
+async function apiCall(url: string) {
+  const authToken = JSON.parse(window.localStorage.getItem('kavita-user') || '{}').token;
+  return api.request.xhr('GET', {
+    url,
+    headers: {
+      Accept: 'application/json',
+      Authorization: `Bearer ${authToken}`,
+    },
+  });
+}
+
+const chapterDetails = {
+  volumeNum: '',
+  chapterNum: '',
+  currentPageNum: '',
+  totalPageNum: '',
+};
+
+const seriesDetails = {
+  seriesTitle: '',
+  seriesID: '',
+  totalPageNum: '',
+  anilistLink: '',
+  kitsuLink: '',
+  malLink: '',
+};
+
+async function getChapterProgress() {
+  const chapterID = window.location.pathname.split('/')[6];
+  const url = `${window.location.origin}/api/reader/get-progress?chapterid=${chapterID}`;
+  const response = await apiCall(url);
+  const responseJSON = JSON.parse(response.responseText);
+  con.log('Chapter Progress Retrieved', response);
+
+  // Handle weird kavita api case where current page number is 1 lesser than what it actually is
+  // unless it is the last page
+  if (parseInt(responseJSON.pageNum) < parseInt(chapterDetails.totalPageNum)) {
+    chapterDetails.currentPageNum = responseJSON.pageNum + 1;
+  } else {
+    chapterDetails.currentPageNum = chapterDetails.totalPageNum;
+  }
+}
+
+export const Kavita: pageInterface = {
+  name: 'Kavita',
+  domain: 'https://demo.kavitareader.com',
+  languages: ['Many'],
+  type: 'manga',
+  isSyncPage(url) {
+    return utils.urlPart(url, 7) === 'manga' && utils.urlParam(url, 'incognito') !== 'true';
+  },
+  isOverviewPage(url) {
+    return utils.urlPart(url, 5) === 'series' && url.split('/').length === 7;
+  },
+  sync: {
+    getTitle(url) {
+      if (!seriesDetails.seriesTitle) throw 'No Title';
+      return seriesDetails.seriesTitle;
+    },
+    getIdentifier(url) {
+      if (!seriesDetails.seriesID) throw 'No series id';
+      return seriesDetails.seriesID;
+    },
+    getOverviewUrl(url) {
+      return window.location.href.split('/').slice(0, 7).join('/');
+    },
+    getEpisode(url) {
+      if (!chapterDetails.chapterNum || !parseInt(chapterDetails.chapterNum))
```

Comment:
This condition triggers on chapterNum = 0. Which I think is not intended as it would mean it will never track volumen only entries.

Reply:
Ah gotcha. I'll copy the Mangadex implementation: https://github.com/MALSync/MALSync/blob/master/src/pages/Mangadex/main.ts#L69